diag_manager/fms_diag_file_object.F90

```diff
@@ -1119,6 +1120,40 @@ subroutine write_time_metadata(this)
 
 end subroutine write_time_metadata
 
+!> \brief Write out the field data to the file
+subroutine write_field_data(this, field_obj, buffer_obj)
+  class(fmsDiagFileContainer_type), intent(in), target   :: this !< The file object
+  type(fmsDiagField_type),                 intent(in), target :: field_obj(:)
+  type(fmsDiagOutputBufferContainer_type), intent(in), target :: buffer_obj(:)
+
+  class(fmsDiagFile_type), pointer     :: diag_file      !< Diag_file object to open
+  class(FmsNetcdfFile_t),  pointer     :: fileobj        !< Fileobj to write to
+  integer                              :: i              !< For do loops
+  integer                              :: field_id       !< The id of the field writing the data to
+
+  diag_file => this%FMS_diag_file
+  fileobj => diag_file%fileobj
+
+  if (diag_file%is_static) then
+    !< Here the file is static so there is no need for the unlimited dimension
+    !! as a variables are static
+    do i = 1, diag_file%number_of_buffers
+      call buffer_obj(diag_file%buffer_ids(i))%write_buffer(fileobj)
+    enddo
+  else
+    do i = 1, diag_file%number_of_buffers
```

Comment:
Can this be done in parallel?

Reply:
I don't think we can do io in parallel, yet(?)

Reply:
This is actually a great response lol